frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2173,6 +2215,22 @@ def encrypt(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
     def equal(self, resulting_type: ConversionType, x: Conversion, y: Conversion) -> Conversion:
         return self.comparison(resulting_type, x, y, accept={Comparison.EQUAL})
 
+    def shift_left(self, x: Conversion, rank: int) -> Conversion:
```

Comment:
I think having this name is dangerous, it's not for the traced `x << 3` operation, not exactly. It'll keep the precision whereas `x << 3` can result in any bit width due to assignment.

I understand the purpose, but the name should not be an existing operation. Maybe we should call it, `shift_left_without_changing_precision`?

Reply:
Renamed